addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -102,11 +103,11 @@ def _process_step_create_lead_and_forward(self, discuss_channel):
             self.env["res.users"].browse(assignable_user_ids)
         )[discuss_channel.livechat_channel_id]
         message = self._process_step_forward_operator(discuss_channel, users=users)
-        operator_partner = discuss_channel.livechat_operator_id
-        if operator_partner != self.chatbot_script_id.operator_partner_id:
-            user = next(user for user in users if user.partner_id == operator_partner)
+
+        if operator_partner != discuss_channel.livechat_operator_id:
+            user = next((user for user in users if user.partner_id == operator_partner), operator_partner.user_id)
```

Comment:
The other changes are no longer necessary in this case, it shouldn't be possible for the `operator_partner` to change if there is no user.